test/CMakeLists.txt

```
@@ -23,6 +23,7 @@ celeritas_add_library(testcel_harness
 target_compile_features(testcel_harness PUBLIC cxx_std_14)
 celeritas_target_link_libraries(testcel_harness
   PUBLIC Celeritas::corecel GTest::GTest
+  PRIVATE ${nlohmann_json_LIBRARIES}
```

Is it really okay to have `PRIVATE` followed by nothing?
What if later on somebody add something after:
```suggestion
  PRIVATE ${nlohmann_json_LIBRARIES}
  OTHERTAG ${othervalues}
```
?

---

I tested before pushing and it does work as expected.

---

Also besides public and private what other tags go there?

---

Beside the many legacy flags there is `INTERFACE` and then there is the new one that will be introduce in cmake v5.22 in a few years :).

> I tested before pushing and it does work as expected.

So you tested something like:
```
celeritas_target_link_libraries(testcel_harness
  PRIVATE ${nlohmann_json_LIBRARIES}
  PUBLIC Celeritas::corecel GTest::GTest
)  
```
or even
```
celeritas_target_link_libraries(testcel_harness
  PRIVATE 
  PUBLIC Celeritas::corecel GTest::GTest
)  
```
right?

---

I didn't flip the lines, I tested the code that's in there. With JSON unavailable, the current code works.

---

Yes, I expect so.  There is nothing to be confused with.  But if you flip the lines or Seth 2030 add a new tag afterwards, then it might (hopefully not) be a different story.

---

Seth 2030 will probably see that there's a possibly-empty variable beforehand, and Philippe 2030 will give it a thorough code review 😉

---

Fair enough ...